### PR TITLE
[ Tensor ] Support NHWC for dot, add/multiply_strided and other ops

### PR DESCRIPTION
**Pull request by adwaith-a**
This PR includes changes of Tensor and TensorDim to support NHWC computation for dot, add_strided, multiply_strided, cat, split, and transpose. It also includes unit tests to evaluate.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by jijoongmoon** [comment]
There are CI issues. Could you check them?

**Review by DonghakPark — COMMENTED** [review]
Please Check Unit test & Doxygen For CI

you can check failed test in [HERE](http://ci.nnstreamer.ai/nntrainer/ci/repo-workers/pr-checker/2238-202307032001270.82738900184631-80868ae501971f718240e4f9728d936b0cedf2b7/report/build_log_2238_ubuntu_output.txt)

File: nntrainer/tensor/tensor.cpp [hunk 1]
```diff
@@ -358,28 +413,70 @@ Tensor &Tensor::multiply_strided(Tensor const &m, Tensor &output,
...
-  if (strides[3] != 1 || m.strides[3] != 1 || output.strides[3] != 1 ||
-      beta != 0.0) {
-    for (unsigned int b = 0; b < batch(); ++b) {
-      for (unsigned int c = 0; c < channel(); ++c) {
-        for (unsigned int h = 0; h < height(); ++h) {
-          for (unsigned int w = 0; w < width(); ++w) {
-            output.addValue(
-              b, c, h, w, getValue(b, c, h, w) * m.getValue(b, c, h, w), beta);
+  NNTR_THROW_IF(!contiguous || !m.contiguous || !output.contiguous,
+                std::invalid_argument)
```
Comment (jijoongmoon) [inline nntrainer/tensor/tensor.cpp:417]: It is strided computation so.. It doesn't need to be contiguous.

Comment (adwaith-a) [reply nntrainer/tensor/tensor.cpp]: Updated. Thanks.

**Review by jijoongmoon — APPROVED** [review]
LGTM. Thanks for the great work.

**Review by skykongkong8 — COMMENTED** [review]
LGTM.
As far as I am concerned, even considering the development of class template instantiation in the future, no defects were found.

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: test/unittest/unittest_nntrainer_tensor.cpp [hunk 2]
```diff
@@ -210,37 +118,47 @@ TEST(nntrainer_Tensor, Tensor_01_p) {
...
-    std::vector<float> tv;
-    for (int j = 0; j < width; ++j) {
-      tv.push_back(i * 2.0 + j);
-    }
-    in.push_back(tv);
-  }
-
-  nntrainer::Tensor tensor = nntrainer::Tensor(in);
-  ASSERT_NE(nullptr, tensor.getData());
+// TEST(nntrainer_Tensor, Tensor_02_p) {
```
Comment (jijoongmoon) [inline test/unittest/unittest_nntrainer_tensor.cpp:121]: Is there any reason for disabling this test?

**Pull request merged**